### DISCUSSION TITLE
Possible FAQ needed, correct combination of importOrderParserPlugins for Angular parameter decorators

### DISCUSSION THREAD

**Discussion by adam-marshall** [trivago/prettier-plugin-sort-imports#122] [Q&A]
If I use parameter decorators e.g. https://angular.io/api/core/Self#usage-notes

I receive the following error
`SyntaxError: Decorators cannot be used to decorate parameters.`

I saw similar closed issues but it is still not clear to me today what the correct configuration would be to allow this, I currently use:

`"importOrderParserPlugins" : [ "typescript", "classProperties", "[\"decorators\", { \"decoratorsBeforeExport\": true }]"]`

I tried a few things but no joy yet.

**Comment by adam-marshall** [comment]
actually, it looks like this isn't covered in https://github.com/trivago/prettier-plugin-sort-imports/blob/master/tests/Angular/imports-with-decorators.js so it might be worth being an actual issue to capture that.

**Reply by adam-marshall** [reply]
I can reproduce this simply by running the tests locally and adding a decorator to a constructor parameter
![Screenshot 2022-01-12 at 10 08 23](https://user-images.githubusercontent.com/2521384/149120239-a3e48ef9-de11-4dbc-9a5c-7f02e5f01b9c.png)

**Comment by ratierd** [comment]
I've had better results in angular using the decorators-legacy plugin instead.

Maybe I'm missing something but I think this config should work
```
importOrderParserPlugins : [ "typescript", "classProperties", "decorators-legacy"]
```

Please tell me if that solves your issue.

Regardless, this is not the first time people report issues with angular projects. So the doc might need some update :sweat_smile:

**Reply by Sebastian-G** [reply]
`SyntaxError: Decorators cannot be used to decorate parameters.`

Still the same issue.

I my case I was trying to format a file containing some decorated constructor injections:
e.g. `constructor(@Optional() @SkipSelf() parentModule: CoreModule) {`

**Comment by laustdeleuran** [comment]
+1, I'm having the exact same issue as @Sebastian-G and @adam-marshall, which effectively renders this plugin unusable to me. I'm on a NestJS project, though, so using Angular doesn't seem to be the issue.
